### PR TITLE
DA1469x counter driver

### PR DESCRIPTION
**Pull request by blauret**
This PR is adding the counter for the Renesas DA1469x microcontroller family

### CODE REVIEW

**Comment by blauret** [comment]
The PR depends on #54263 and all test should be passing when it is merged.

File: drivers/counter/counter_smartbond_timer.c [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+static int counter_smartbond_set_alarm(const struct device *dev, uint8_t chan,
+				       const struct counter_alarm_cfg *alarm_cfg)
+{
+	const struct counter_smartbond_config *config = dev->config;
+	struct counter_smartbond_data *data = dev->data;
+	TIMER2_Type *timer = config->timer;
+
+	__IOM uint32_t *timer_clear_irq_reg = ((TIMER_Type *)timer) == TIMER
+						      ? &((TIMER_Type *)timer)->TIMER_CLEAR_IRQ_REG
+						      : &timer->TIMER2_CLEAR_IRQ_REG;
```
Comment (andrzej-kaczmarek) [inline drivers/counter/counter_smartbond_timer.c:108]: this is "calculated" more than once in driver (`counter_smartbond_irq_handler`) so perhaps make it a helper func

File: drivers/counter/counter_smartbond_timer.c [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+static int counter_smartbond_set_top_value(const struct device *dev,
+					   const struct counter_top_cfg *cfg)
+{
+	ARG_UNUSED(dev);
+
+	if (cfg->ticks != 0xFFFFFF) {
+		return -ENOTSUP;
+	}
+
+	return 0;
```
Comment (andrzej-kaczmarek) [inline drivers/counter/counter_smartbond_timer.c:208]: I think it would make more sense to always return `-ENOTSUP` since top value is fixed

**Review by andrzej-kaczmarek — CHANGES_REQUESTED** [review]

File: drivers/counter/counter_smartbond_timer.c (hunk 1 above)
Comment (blauret) [reply drivers/counter/counter_smartbond_timer.c]: Refactored in one function

**Pull request closed without merging**